src/rgw/driver/rados/rgw_tools.cc

```diff
@@ -164,7 +164,7 @@ int rgw_delete_system_obj(const DoutPrefixProvider *dpp,
 
 int rgw_rados_operate(const DoutPrefixProvider *dpp, librados::IoCtx& ioctx, const std::string& oid,
                       librados::ObjectReadOperation *op, bufferlist* pbl,
-                      optional_yield y, int flags)
+                      optional_yield y, int flags, const jspan_context* trace_info)
```

Comment:
Can it be ever `nullptr` at this layer? If not, then perhaps a `const jspan_context&` would express this better.

Reply:
it is `nullptr` by default in many function calls (to avoid many code changes where traces are not used at all). 
in this case, the function decleration is:
```cpp
int rgw_rados_operate(const DoutPrefixProvider *dpp, librados::IoCtx& ioctx, const std::string& oid,                                                                                                                                          
                      librados::ObjectReadOperation *op, bufferlist* pbl,
                      optional_yield y, int flags = 0, const jspan_context* trace_info = nullptr);

```
also, in `Objecter::_prepare_osd_op()` (invoked from the above) there is a line that check if `op->otel_trace` is nullptr.